src/main/java/group2/cms/service/AuthorService.java

```diff
@@ -23,47 +27,66 @@
     @Autowired
     private AuthorRepository authorRepository;
 
-    public Author addAuthor(Long userID, Long paperID){
+    public AuthorDTO addAuthor(AuthorDTO authorDTO){
+        var authorData = converter.dtoToEntity(authorDTO);
+        var userID = authorData.getId();
+        var paperID = authorData.getPaper().getId();
         var paperUploader = userRepository.findById(userID).orElseThrow(
                 () -> new InvalidIDException("Invalid User ID: " + userID)
         );
         var uploadedPaper = paperRepository.findById(paperID).orElseThrow(
                 () -> new InvalidIDException("Invalid Paper ID: " + paperID)
         );
 
-        return authorRepository.save(new Author(paperUploader, uploadedPaper));
+        var addedAuthor =  authorRepository.save(new Author(paperUploader, uploadedPaper));
+        return converter.entityToDto(addedAuthor);
     }
 
-    public void deleteAuthor(Long authorID){
+    public void deleteAuthor(Long authorID)
+    {
         authorRepository.deleteById(authorID);
     }
 
-    public List<Author> getAllAuthors(){
-        return authorRepository.findAll();
+    public AuthorsDTO getAllAuthors(){
+        var authorsDTO = new AuthorsDTO();
+        authorRepository.findAll().stream()
+                .map(converter::entityToDto)
+                .forEach(authorsDTO::addDTO);
+        return authorsDTO;
     }
 
-    public List<Author> getAuthorsOfPaper(Long paperID){
+    public AuthorsDTO getAuthorsOfPaper(AuthorDTO authorDTO){
+        var paperID = converter.dtoToEntity(authorDTO).getPaper().getId();
         var paper = paperRepository.findById(paperID).orElseThrow(
                 () -> new InvalidIDException("Invalid Paper ID: " + paperID)
         );
-        return authorRepository.findAuthorsByPaper(paper);
+        var authorsDTO = new AuthorsDTO();
```

Comment:
I do not think that we should build the DTO in the service. You should build that array and then convert it to the dto using the converter

Reply:
I will try to modify the converter to work also with EntitiesDTO.